universal-login-sdk/lib/core/services/MineableFactory.ts

```diff
@@ -12,4 +13,10 @@ export class MineableFactory {
   protected isMined(state: MineableState) {
     return state === 'Error' || state === 'Success';
   }
+
+  protected hasTransactionHash(status: MineableStatus) {
+    return ['Pending', 'Success', 'Error'].includes(status.state) &&
+      status.transactionHash !== null &&
+      status.transactionHash !== undefined;
```

Comment:
Maybe if transaction hash is missing in this case (status state is Pending, Success or Error) throw error? (ensureNotNull)

Reply:
Done.